pkg/clients/edgeconnect/client.go

```diff
@@ -0,0 +1,278 @@
+package edgeconnect
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type client struct {
+	ctx     context.Context
+	baseURL string
+	clientcredentials.Config
+}
+
+// Option can be passed to NewClient and customizes the created client instance.
+type Option func(client *client)
+
+// NewClient creates a REST client for the given API base URL
+// opts can be used to customize the created client, entries must not be nil.
+func NewClient(clientID, clientSecret string, options ...Option) (Client, error) {
+	c := &client{
+		Config: clientcredentials.Config{
+			ClientID:     clientID,
+			ClientSecret: clientSecret,
+			Scopes:       DefaultOauthScopes,
+			TokenURL:     DefaultTokenURL,
+		},
+	}
+
+	for _, opt := range options {
+		opt(c)
+	}
+
+	return c, nil
+}
+
+func NewClientWithProxy(clientID, clientSecret, proxy string) (Client, error) {
```

Comment:
In the ticket, we highlighted having such an option; it's impossible because it's not implemented in CR. I decided to add it to keep an example of how to do it.

Reply:
@StefanHauth I've added a comment